src/ast.rs

```diff
@@ -5,28 +5,23 @@ use std::collections::HashMap;
 #[derive(Clone, Debug, PartialEq)]
 pub struct Tag {
     pub name: String,
+    pub properties: Vec<String>,
 }
 
 impl Tag {
-    pub fn new<T: ToString>(name: T) -> Tag {
+    pub fn new<T: ToString>(name: T, properties: Vec<String>) -> Tag {
         Tag {
             name: name.to_string(),
+            properties,
         }
     }
 }
 
 #[derive(Clone, Debug, PartialEq)]
 pub struct TopAst {
-    pub tag: Option<Tag>,
     pub ast: TopAstVariant,
 }
 
-impl TopAst {
-    pub fn new(tag: Option<Tag>, ast: TopAstVariant) -> TopAst {
-        TopAst { tag, ast }
-    }
-}
-
```

Comment:
Remove this concept about tag only appearing with top ast, since class member also needs to be checked.